test/bitonic_sort_test.dart

```diff
@@ -0,0 +1,113 @@
+import 'package:rohd_hcl/rohd_hcl.dart';
+import 'package:test/test.dart';
+
+Future<void> main() async {
+  tearDown(() async {
+    await Simulator.reset();
+  });
+
+  group('Bitonic Sort', () {
+    test(
+        'should return the sorted results in ascending order '
+        'given descending order', () async {
+      const dataWidth = 8;
+      const direction = 1;
+
+      final clk = SimpleClockGenerator(10).clk;
+      final reset = Logic(name: 'reset');
+
+      final x = <Logic>[
+        Const(16, width: dataWidth),
+        Const(15, width: dataWidth),
+        Const(14, width: dataWidth),
+        Const(13, width: dataWidth),
+        Const(12, width: dataWidth),
+        Const(11, width: dataWidth),
+        Const(10, width: dataWidth),
+        Const(9, width: dataWidth),
+        Const(8, width: dataWidth),
+        Const(7, width: dataWidth),
+        Const(6, width: dataWidth),
+        Const(5, width: dataWidth),
+        Const(4, width: dataWidth),
+        Const(3, width: dataWidth),
+        Const(2, width: dataWidth),
+        Const(1, width: dataWidth),
+      ];
+
+      final topMod = BitonicSort(clk, reset, x, direction, name: 'top_level');
+      await topMod.build();
+
+      reset.inject(0);
+
+      Simulator.registerAction(100, () {
+        for (var i = 0; i < topMod.sorted.length; i++) {
+          expect(topMod.sorted[i].value.toInt(), i + 1);
+        }
+      });
+
+      Simulator.setMaxSimTime(100);
+      // WaveDumper(topMod, outputPath: 'lib/src/sort/recursive_list.vcd');
+
+      await Simulator.run();
+    });
+
+    test(
+        'should return the sorted results in descending order '
+        'given ascending order', () async {
+      const dataWidth = 8;
+      const direction = 0;
+
+      final clk = SimpleClockGenerator(10).clk;
+      final reset = Logic(name: 'reset');
+
+      final x = <Logic>[
+        Const(1, width: dataWidth),
+        Const(2, width: dataWidth),
+        Const(3, width: dataWidth),
+        Const(4, width: dataWidth),
+        Const(5, width: dataWidth),
+        Const(6, width: dataWidth),
+        Const(7, width: dataWidth),
+        Const(8, width: dataWidth),
+        Const(9, width: dataWidth),
+        Const(10, width: dataWidth),
+        Const(11, width: dataWidth),
+        Const(12, width: dataWidth),
+        Const(13, width: dataWidth),
+        Const(14, width: dataWidth),
+        Const(15, width: dataWidth),
+        Const(16, width: dataWidth),
+      ];
+
+      final topMod = BitonicSort(clk, reset, x, direction, name: 'top_level');
+      await topMod.build();
+
+      reset.inject(0);
+
+      Simulator.registerAction(100, () {
+        for (var i = 0; i < topMod.sorted.length; i++) {
+          expect(topMod.sorted[i].value.toInt(), topMod.sorted.length - i);
+        }
+      });
+
+      Simulator.setMaxSimTime(100);
+      // WaveDumper(topMod, outputPath: 'lib/src/sort/recursive_list.vcd');
+
+      await Simulator.run();
+    });
+  });
```

Comment:
Add a test where we push different arrays in every cycle, and check that the output changes the expected number of cycles later (basically demonstrate that this can be used as a pipelined sorter).